### DISCUSSION TITLE
Windows - run a PowerShell script

### DISCUSSION THREAD

**Discussion by MSP-Greg** [community/community#26116] [Actions]
<p>Trying to move CI from AppVeyor to Actions, which used PowerShell files for build and test.</p>
<p>Reviewed <a href="https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idstepsrunshell" rel="nofollow noopener">https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idstepsrunshell</a>, which I found without examples.  So, stupid question - how to run a ps1 file in an Actions step?</p>

**Comment by rajsite** [comment]
<p>I was also trying to figure out how to use the shell option to run a powershell script. The doc shows the following definition:</p>
<pre><code>jobs.&lt;job_id&gt;.steps.run.shell
</code></pre>
<p>So I tried to make a sub-property of run (which I don’t think is valid yml, but the editor doesn’t do any validation so it’s hard to tell):</p>
<pre><code>name: CI

on:
  push:
    branches:
    - master

jobs:
  build:

    runs-on: windows-2019

    steps:
    - uses: actions/checkout@v1
    - name: Run a one-line script
      run: .\MyScripts\install.ps1
        shell: powershell
</code></pre>
<p>Which errored with:</p>
<pre><code>yaml: line 17: mapping values are not allowed in this context
</code></pre>
<p>I gave up and just used powershell from the cmd command instead of trying to change the shell:</p>
<pre><code>name: CI

on:
  push:
    branches:
    - master

jobs:
  build:

    runs-on: windows-2019

    steps:
    - uses: actions/checkout@v1
    - name: Run a one-line script
      run: powershell .\MyScripts\install.ps1
</code></pre>
<p>Hopefully some examples can be added to that page to show how to actually use the shell parameter</p>

**Comment by damccorm** [comment]
<p>Try:</p>
<pre><code>name: CI

on:
  push:
    branches:
    - master

jobs:
  build:

    runs-on: windows-2019

    steps:
    - uses: actions/checkout@v1
    - name: Run a one-line script
      run: .\MyScripts\install.ps1
      shell: powershell
</code></pre>
<p>Its working for me - <a href="https://github.com/damccorm/actions-playground/pull/9" rel="nofollow noopener">https://github.com/damccorm/actions-playground/pull/9</a> - I think this is just a docs issue</p>
<p>cc <a class="mention" href="/rachmari">@rachmari</a> </p>

**Comment by rachmari** [comment, marked as answer]
<p>Hey there, thanks for pointing this out. It was indeed a documentation bug. We’ve fixed the documentation and added examples for using different shells! Let me know if there is still some clarification we can make in the documentation. </p>
<p>You’ll find the updated documentation here:<br>
<a href="https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idstepsrun" rel="nofollow noopener">https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idstepsrun</a></p>
